test/mock/service.go

```diff
@@ -79,6 +115,9 @@ func MakeInstance(service *model.Service, port *model.Port, version int) *model.
 // MakeIP creates a fake IP address for a service and instance version
 func MakeIP(service *model.Service, version int) string {
 	ip := net.ParseIP(service.Address).To4()
+	if ip == nil {
```

Comment:
as a general istio user wishing to use an external service - are they required to know the hostname (google.com) and the IP it resolves to?  Or from a coding standpoint then, if an external service is specified how do we populate the IP field, or would we just leave IP/Address as an empty string in this case? And if left empty, we would still need this `ip == nil` check in the code or else the next few lines trigger an index out of bounds exception